modules/45_carbonprice/NDC/datainput.gms

```diff
@@ -15,64 +16,92 @@ pm_taxCO2eq("2020",regi)= pm_taxCO2eq("2020",regi) * sm_DptCO2_2_TDpGtC;
-$if "%cm_NDC_version%" == "2018_uncond" $include "./modules/45_carbonprice/NDC/input/set_regi2030_NDC2018_uncond.cs4r"
-$offdelim
-  /             ;
-  
+*** parameters for selecting NDC years
+Scalar p45_ignore_NDC_before             "NDC targets before this years are ignored, for example to exclude 2030 targets" /2020/;
+Scalar p45_ignore_NDC_after              "NDC targets after  this years are ignored, for example to exclude 2050 net zero targets" /2030/;
+Scalar p45_min_ratio_of_coverage_to_max  "only targets whose coverage is this times p45_best_NDC_coverage are considered. Use 1 for only best." /1.0/;
+Scalar p45_use_single_year_close_to      "if 0: use all. If > 0: use only one single NDC target per country closest to this year (use 2030.4 to prefer 2030 over 2035 over 2025)" /2030.4/;
+
+Set p45_NDC_year_set(ttot,all_regi)               "YES for years whose NDC targets is used";
+Parameter p45_best_NDC_coverage(all_regi)         "highest coverage of NDC targets within region";
+Parameter p45_distance_to_optyear(ttot,all_regi)  "distance to p45_use_single_year_close_to to favor years in case of multiple equally good targets";
+Parameter p45_min_distance_to_optyear(all_regi)   "minimal distance to p45_use_single_year_close_to per region";
+
+p45_best_NDC_coverage(regi) = smax(ttot$(ttot.val <= p45_ignore_NDC_after AND ttot.val >= p45_ignore_NDC_before), p45_2005share_target(ttot,regi));
+display p45_best_NDC_coverage;
+
+p45_NDC_year_set(ttot,regi)$(ttot.val <= p45_ignore_NDC_after AND ttot.val >= p45_ignore_NDC_before) = p45_2005share_target(ttot,regi) >= p45_min_ratio_of_coverage_to_max * p45_best_NDC_coverage(regi);
+
+if(p45_use_single_year_close_to > 0,
+  p45_distance_to_optyear(p45_NDC_year_set(ttot,regi)) = abs(ttot.val - p45_use_single_year_close_to);
+  p45_min_distance_to_optyear(regi) = smin(ttot$(p45_NDC_year_set(ttot,regi)), p45_distance_to_optyear(ttot,regi));
+  p45_NDC_year_set(ttot,regi) = p45_distance_to_optyear(ttot,regi) = p45_min_distance_to_optyear(regi);
+);
+
+*** first and last NDC year as a number
+Parameter p45_first_NDC_year(all_regi) "last year with NDC coverage within region";
+p45_first_NDC_year(regi) = smin( p45_NDC_year_set(ttot, regi), ttot.val );
+Parameter p45_last_NDC_year(all_regi)  "last year with NDC coverage within region";
+p45_last_NDC_year(regi)  = smax( p45_NDC_year_set(ttot, regi), ttot.val );
+
+display p45_NDC_year_set,p45_first_NDC_year,p45_last_NDC_year;
+
+*** adjust reduction value for LAM based on the assumption that Brazilian reduction targets are only from landuse.
+*** the adjustment were taken such that Brazil is assumed to maintain its 2015 non-landuse emissions
+
+p45_factor_targetyear(ttot,regi)$(sameas(regi,"LAM") AND sameas(ttot,"2025")) = p45_factor_targetyear(ttot,regi) + 0.46;
+p45_factor_targetyear(ttot,regi)$(sameas(regi,"LAM") AND sameas(ttot,"2030")) = p45_factor_targetyear(ttot,regi) + 0.205;
```

Comment:
where are the numbers from? Do you really need this region name here and could not find a general formulation?

Reply:
the assumption is that Brazil's emission reduction is predominantly from changes in landuse emissions. I took `ghgfactor` from `[calcEmiTarget.R](https://github.com/pik-piam/mrremind/blob/master/R/calcEmiTarget.R)` (mrremind), calculated the share of Brazil within the LAM target and then how much the ghgfactor for LAM had to be increased to keep Brazil's emissions constant. This information cannot be obtained after the aggregation process, and I preferred to enter it directly here instead of adapting the input data. This procedure is of relevance particularly for LAM as all other regions have much lower (changes in) landuse emissions. What complicates a more general solution is that the differentiation between landuse and other emissions is not stated in the NDC documents.

Reply:
could you introduce a rule that - based on the share of landuse emissions - corrects ghgfactors to keep emissions constant. This rule might not affect all regions but mainly "LAM"?
At least: give an explanation for the values: nobody (including yourself) will be able to understand this in 6 months :-)

Reply:
I have chosen the "at least" option and added a detailed explanation how this value was derived and why, and also intercepted the case of 2018_uncond, where Brazil does not actually have a target. More general solution certainly is desirable, but Brazil is indeed a special case with its very high share of emissions related to landuse.